### ISSUE TITLE
Exception with IntelliJ IDEA 2019.3 EAP

### ISSUE THREAD

**Issue by kedar-joshi** [dubreuia/intellij-plugin-save-actions#259]
**Describe the bug**
I am seeing following exception multiple times throughout the time I'm working on any project. Apparently, after this commit https://github.com/JetBrains/intellij-community/commit/eb5cf4a8326f08a61dbbcf126ae9b689faa88e5c constructors need to be annotated with `@NonInjectable`.

```
com.intellij.diagnostic.PluginException: getComponentAdapterOfType is used to get com.dubreuia.model.Storage (requestorClass=com.dubreuia.model.Storage, requestorConstructor=public com.dubreuia.model.Storage(com.dubreuia.model.Storage)).

Probably constructor should be marked as NonInjectable. [Plugin: com.dubreuia]
	at com.intellij.serviceContainer.ConstructorParameterResolver.findTargetAdapter(ConstructorParameterResolver.kt:109)
	at com.intellij.serviceContainer.ConstructorParameterResolver.isResolvable(ConstructorParameterResolver.kt:37)
	at com.intellij.serviceContainer.ConstructorInjectionKt.getGreediestSatisfiableConstructor(constructorInjection.kt:106)
	at com.intellij.serviceContainer.ConstructorInjectionKt.instantiateUsingPicoContainer(constructorInjection.kt:36)
	at com.intellij.serviceContainer.PlatformComponentManagerImpl.instantiateClassWithConstructorInjection(PlatformComponentManagerImpl.kt:500)
	at com.intellij.serviceContainer.ServiceComponentAdapter.createAndInitialize(ServiceComponentAdapter.kt:52)
	at com.intellij.serviceContainer.ServiceComponentAdapter.doCreateInstance(ServiceComponentAdapter.kt:39)
	at com.intellij.serviceContainer.BaseComponentAdapter.getInstanceUncached(BaseComponentAdapter.kt:100)
	at com.intellij.serviceContainer.BaseComponentAdapter.getInstance(BaseComponentAdapter.kt:72)
	at com.intellij.serviceContainer.BaseComponentAdapter.getInstance$default(BaseComponentAdapter.kt:65)
	at com.intellij.serviceContainer.PlatformComponentManagerImpl.getService(PlatformComponentManagerImpl.kt:288)
	at com.intellij.openapi.components.ServiceManager.getService(ServiceManager.java:22)
...
```

Full log is attached.
[2019.3-EAP.txt](https://github.com/dubreuia/intellij-plugin-save-actions/files/3639480/2019.3-EAP.txt)

**What triggered the plugin**
CTRL+SHIFT+S 

**Versions**
```
IntelliJ IDEA 2019.3 EAP (Ultimate Edition)
Build #IU-193.3519.25, built on September 18, 2019
IntelliJ IDEA EAP User
Expiration date: October 18, 2019
Runtime version: 11.0.4+10-b480.2 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 4029M
Cores: 8
Registry: 
Non-Bundled Plugins: Struts Assistant, com.codota.csp.intellij, com.dubreuia, com.intellij.plugins.html.instantEditing, org.intellij.scala, org.jetbrains.plugins.vue
```

**Comment by dubreuia** [comment]
I love breaking changes from Jetbrains. Thanks @kedar-joshi, I'm looking into it.

**Comment by dubreuia** [comment]
LGTM, I'll release the plugin ASAP

**Comment by dubreuia** [comment]
@kedar-joshi Where did you find the jar version `193.3519.25-EAP-SNAPSHOT` at build.gradle:22? I'm not a big fan of having a dependency on a SNAPSHOT version in our build, but I don't know how to find the other versions.

**Comment by kedar-joshi** [comment]
> @kedar-joshi Where did you find the jar version `193.3519.25-EAP-SNAPSHOT` at build.gradle:22? I'm not a big fan of having a dependency on a SNAPSHOT version in our build, but I don't know how to find the other versions.

@dubreuia I saw the version format for EAP here https://github.com/JetBrains/gradle-intellij-plugin/issues/378 and extrapolated for current EAP version.

**Comment by dubreuia** [comment]
On master we can keep the latest EAP release, I'm not releasing on master. I'll backport the commit content to the proper branches.

I'm found this:

- https://www.jetbrains.com/intellij-repository/snapshots
    - 193.3519.25 (LATEST-EAP-SNAPSHOT)
- https://www.jetbrains.com/intellij-repository/releases
    - 192.6603.28 (2019.2.2)

**Comment by dubreuia** [comment]
Note to self, we have a SNAPSHOT dependency (this is awful) because:

> Yes, eaps now are always snapshots.

See https://github.com/JetBrains/gradle-intellij-plugin/issues/378#issuecomment-476125062

**Comment by dubreuia** [comment]
Released in 1.6.0: https://plugins.jetbrains.com/plugin/7642-save-actions/versions
